src/pycell.rs

```diff
@@ -638,6 +637,34 @@ where
     }
 }
 
+impl<'p, T: PyClass> PyRef<'p, T> {
+    /// Returns the raw FFI pointer represented by self.
+    ///
+    /// # Safety
+    ///
+    /// Callers are responsible for ensuring that the pointer does not outlive self.
+    ///
+    /// The reference is borrowed; callers should not decrease the reference count
+    /// when they are finished with the pointer.
+    #[inline]
+    pub fn as_ptr(&self) -> *mut ffi::PyObject {
```

Comment:
I wonder if `PyRef` should `Deref` into `PyAny` via forwarding to its inner `PyCell` field instead as that more generally useful?

Reply:
Unfortunately `PyRef<T>` already implements `Derek<Target = T>`, which is extremely useful 😉

Reply:
Indeed. How about `AsRef<PyAny>` as alternative? (Personally, I aim for `AsRef<T>` as well if I have `Deref<Target=T>` which is usually not problematic as one can have multiple `AsRef` impls.)

(My reasoning here is that I would like to centralize the pointer conversion as much as possible at `Py` and `PyAny`.)

Reply:
I'll give that a try and see what breaks 😅

Reply:
So I started down this road; here's a reason not to add `AsRef<PyAny>` for now:

The implementation would be

```
impl<T: PyClass> AsRef<PyAny> for PyRef<'_, T> {
    fn as_ref(&self) -> &PyAny {
        unsafe { self.py().from_borrowed_ptr(self.as_ptr()) }
    }
}
```

and the problem is that the `&PyAny` there has a much shortened lifetime, it's the duration of the `.as_ref()` borrow rather than the GIL lifetime `'p` attached to the `PyRef`.

Given this isn't an ideal UX, I would like to take the position that we don't implement `AsRef<PyAny>` for `PyRef` or `PyRefMut` for now. Can I request we move ahead here by just keeping the inherent methods and then perhaps reconsider once we have migrated to `Py<'py, PyAny>`?

Presumably then we can have the following implementation which would have a better lifetime

```
impl<'p, T: PyClass> AsRef<Py<'p, PyAny>> for PyRef<'p, T> {
    fn as_ref(&self) -> &Py<'p, PyAny> {
        // self.inner would be &'p Py<'p, PyCell<T>> or similar, I imagine
        self.inner.as_ref()
    }
}
```

(In the long run we might want to have `PyRef<'py, 'cell, T>` or maybe `Py<'py, PyRef<'cell, T>>` to capture that there are two distinct lifetimes, that of the cell borrow and that of the GIL lifetime. Though that's an an API design question for another time.)